src/core.rs

```diff
@@ -174,13 +180,28 @@ fn ch_width(ch: char) -> usize {
 /// [CJK characters]: https://en.wikipedia.org/wiki/CJK_characters
 /// [emoji modifier sequences]: https://unicode.org/emoji/charts/full-emoji-modifiers.html
 pub fn display_width(text: &str) -> usize {
+    display_width_with_tab(text, 0)
+}
+
+/// Compute the display width of `text` in the same was as
+/// `display_width`, while allowing you to customize how wide
+/// a tab character will be considered to be.
+///
+/// # Examples
+///
+/// ```
+/// use textwrap::core::display_width_with_tab;
+///
+/// assert_eq!(display_width_with_tab("Café \t Plain", 4), 15);
+/// ```
+pub fn display_width_with_tab(text: &str, tab_width: u8) -> usize {
```

Comment:
This is one thing that I don't like too much, but I could not find a way around without changing the public API. Having two methods, one just calling the other with a default parameter, doesn't feel very good, but I can't think of how else to implement this functionality.